File: mobile/lib/features/detection/nutrient_deficiency_controller.dart
import '../../core/image/picked_image.dart';
import '../../models/nutrient_deficiency.dart';
import '../../services/nutrient_deficiency_api.dart';
import 'detection_controller.dart';

/// Nutrient-deficiency detection controller bound to
/// `POST /api/nutrient-deficiency/detect`.
class NutrientDeficiencyController
    extends DetectionController<NutrientDeficiencyResult> {
  NutrientDeficiencyController(NutrientDeficiencyApi api, ImagePickerFn pickImage)
      : super(
          runner: ({
            required List<int> imageBytes,
            required String imageName,
            String? cropName,
          }) =>
              api.detect(
                imageBytes: imageBytes,
                imageName: imageName,
                cropName: cropName,
              ),
          pickImage: pickImage,
        );
}

File: config/core/controllers/nutrient_deficiency_controller.py
"""
KisanAI OS
Nutrient Deficiency Detection Controller

Thin controller for the AI nutrient-deficiency detection feature.
"""

from config.core.services.nutrient_deficiency_service import (
    NutrientDeficiencyService,
)


class NutrientDeficiencyController:
    """Nutrient Deficiency Detection Controller"""

    def __init__(self, service=None):
        self.service = service or NutrientDeficiencyService()

    def detect(self, file, crop_name=None):
        return self.service.detect(file, crop_name)
